mmseg/core/evaluation/metrics.py

```diff
@@ -49,27 +54,28 @@ def intersect_and_union(pred_label,
     label = label[mask]
 
     intersect = pred_label[pred_label == label]
-    area_intersect, _ = np.histogram(
-        intersect, bins=np.arange(num_classes + 1))
-    area_pred_label, _ = np.histogram(
-        pred_label, bins=np.arange(num_classes + 1))
-    area_label, _ = np.histogram(label, bins=np.arange(num_classes + 1))
+    area_intersect = torch.histc(
+        intersect.float(), bins=(num_classes), min=0, max=num_classes)
+    area_pred_label = torch.histc(
+        pred_label.float(), bins=(num_classes), min=0, max=num_classes)
+    area_label = torch.histc(
+        label.float(), bins=(num_classes), min=0, max=num_classes)
     area_union = area_pred_label + area_label - area_intersect
-
     return area_intersect, area_union, area_pred_label, area_label
 
 
-def total_intersect_and_union(results,
-                              gt_seg_maps,
-                              num_classes,
-                              ignore_index,
+def total_intersect_and_union(results: list,
+                              gt_seg_maps: list,
+                              num_classes: int,
+                              ignore_index: int,
                               label_map=dict(),
                               reduce_zero_label=False):
     """Calculate Total Intersection and Union.
 
     Args:
         results (list[ndarray]): List of prediction segmentation maps.
-        gt_seg_maps (list[ndarray]): list of ground truth segmentation maps.
+        gt_seg_maps (list[ndarray]): list of ground truth segmentation
+        maps.
```

Comment:
```suggestion
        gt_seg_maps (list[ndarray]): list of ground truth segmentation
               maps.
```

Reply:
We may update the doc string since `results` and `gt_seg_maps` could be list of str now.